File: vinzi.jsonMgt/src/main/java/vinzi/jsonMgt/hib/TrackInfo.java
package vinzi.jsonMgt.hib;

public class TrackInfo {
    public Long id;
    public Long getId() { return id;}
    protected void setId(Long id)  {this.id = id; };


    public String file_location;
    public String getFile_location() { return file_location;}
    public void setFile_location(String file_location)  {this.file_location = file_location; };


    public String track_name;
    public String getTrack_name() { return track_name;}
    public void setTrack_name(String track_name)  {this.track_name = track_name; };


    public TrackInfo() {};

    public TrackInfo(Long id, String file_location, String track_name) {
	this.id = id;
	this.file_location = file_location;
	this.track_name = track_name;
	return;};

   public String toString() {
        return " id=" + this.id + " file_location=" + this.file_location + " track_name=" + this.track_name;};

};

File: vinzi.jsonMgt/src/main/clojure/vinzi/jsonMgt/hib/clj/TrackInfo.clj
(ns vinzi.jsonMgt.hib.clj.TrackInfo
    (:use vinzi.hib-connect.globals)
    (:import vinzi.jsonMgt.hib.TrackInfo))

(defrecord TrackInfo_clj [id file_location track_name])

(defn create-TrackInfo [file_location track_name]
  ;; TO DO:  insert type-checking (java-objects are strictly typed)
  (TrackInfo_clj. nil file_location track_name))

(defn clone-TrackInfo [cRec]
  ;; TO DO:  insert type-checking cRec should be same type/class as target
  (create-TrackInfo (:file_location cRec) (:track_name cRec)))

(defn TrackInfo-to-java [cRec]
	(TrackInfo. (:id cRec) (:file_location cRec) (:track_name cRec) ))


(add-to-java (TrackInfo_clj. 0 1 2) TrackInfo-to-java)

(defn TrackInfo-to-clj [jRec]
	(TrackInfo_clj. (.getId jRec) (.getFile_location jRec) (.getTrack_name jRec) ))


(add-to-clj (TrackInfo.) TrackInfo-to-clj)

